# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.cpp
# Commit: a1def130c105
# Full Hash: a1def130c1053111631562df9dbf3545356bc448
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2022-01-28 04:41:16
# Regressor Bug: 1742655
# File Overlap Count: 2
# Description:
#   Bug 1742655 - Remove NotifyOutputData from NativeInputTrack r=pehrsons
#   
#   Depends on D134532
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134627
# ==============================================================================

diff -r b0a8837110db -r a1def130c105 dom/media/MediaTrackGraph.cpp
--- a/dom/media/MediaTrackGraph.cpp	Thu Jan 27 22:29:34 2022 +0000
+++ b/dom/media/MediaTrackGraph.cpp	Thu Jan 27 22:29:34 2022 +0000
@@ -21,6 +21,7 @@
 #include "AudioCaptureTrack.h"
 #include "AudioNodeTrack.h"
 #include "AudioNodeExternalInputTrack.h"
+#include "MediaEngineWebRTCAudio.h"
 #include "MediaTrackListener.h"
 #include "mozilla/dom/BaseAudioContextBinding.h"
 #include "mozilla/dom/WorkletThread.h"
@@ -888,9 +889,11 @@
   }
   MOZ_ASSERT(mNativeInputTrack->mDeviceId == mInputDeviceID);
 
-  // When/if we decide to support multiple input devices per graph, this needs
-  // to loop over them.
-  mNativeInputTrack->NotifyOutputData(this, aBuffer, aFrames, aRate, aChannels);
+  for (const auto& track : mTracks) {
+    if (const auto& t = track->AsAudioInputTrack(); t) {
+      t->NotifyOutputData(this, aBuffer, aFrames, aRate, aChannels);
+    }
+  }
 }
 
 void MediaTrackGraphImpl::NotifyInputStopped() {